falcon/util/uri.py

```diff
@@ -264,7 +274,7 @@ def parse_query_string(query_string, keep_blank=False, csv=True):
     # both short and long query strings. Tested on CPython 3.4.
     for field in query_string.split('&'):
         k, _, v = field.partition('=')
-        if not (v or keep_blank):
+        if not v and (not keep_blank or not k):
```

Comment:
Nice catch! One suggestion, though, would be to refactor the statement to reduce the number of boolean operations, e.g.:

```py
if not (v or (keep_blank and k)):
    continue
```

Reply:
(not a big deal; if the version as-is seems to be slightly faster in the common case I'm not opposed to keeping it)

Reply:
I'm admittedly a bit divided here... As discussed on Gitter, estimating that having a value is the most common case, my version actually seems to be outperforming anything else (CPython 3.8):
```python


>>> import timeit
>>> k = 'key'; v = 'value'; keep_blank = True
>>> timeit.timeit(lambda: not (v or (keep_blank and k)), number=int(1e8))
5.702389151003445
>>> timeit.timeit(lambda: not (v or (keep_blank and k)), number=int(1e8))
5.684955962999084
>>> timeit.timeit(lambda: not (v or (keep_blank and k)), number=int(1e8))
5.685243817999435
>>> timeit.timeit(lambda: not v and (not keep_blank or not k), number=int(1e8))
5.500417819006543
>>> timeit.timeit(lambda: not v and (not keep_blank or not k), number=int(1e8))
5.490288328001043
>>> timeit.timeit(lambda: not v and (not keep_blank or not k), number=int(1e8))
5.471835395997914
```